### PR TITLE
new features

### PR DESCRIPTION
**Pull request by aaoferreira**

### CODE REVIEW

**Review by MerlinEgalite — COMMENTED** [review]
Overall LGTM

File: src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    ) {
+        moneyMarketId = _moneyMarketId;
+        positionFactory = _positionFactory;
+        morpho = _morpho;
+        reverseLookup = _reverseLookup;
+    }
+
+    // ====== IMoneyMarketView =======
+
+    function balances(PositionId positionId, IERC20, IERC20) public virtual override returns (Balances memory balances_) {
```
[the remark this answers is not in the document]
Comment (ultrasecreth) [reply src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol:41]: our `balance` function is non view from the interface as other markets (cough Compound) forced us, we don't really mind anyway cause this contract is not meant to be called on-chain, it's a GUI/stats oriented contract

File: src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    function liquidity(PositionId positionId, IERC20 collateralAsset, IERC20)
+        public
+        view
+        virtual
+        returns (uint256 borrowing, uint256 lending)
+    {
+        Market memory market = morpho.market(reverseLookup.marketId(positionId.getPayload()));
+        borrowing = market.totalSupplyAssets - market.totalBorrowAssets;
+        lending = collateralAsset.totalSupply();
```
[the remark this answers is not in the document]
Comment (ultrasecreth) [reply src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol:78]: maybe we got it wrong... @aaoferreira we may need to revisit this one

File: src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    function rates(PositionId positionId, IERC20, IERC20) public view virtual returns (uint256 borrowing, uint256 lending) {
+        Id marketId = reverseLookup.marketId(positionId.getPayload());
+        MarketParams memory params = morpho.idToMarketParams(marketId);
+
+        borrowing = params.irm.borrowRateView(params, morpho.market(marketId));
+        lending = 0;
```
[the remark this answers is not in the document]
Comment (ultrasecreth) [reply src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol:86]: Due to the nature of Morpho Blue, there's no real lending, the wording is maintained for consistency, but what we do is `supplyCollateral` so there's no actual lending and hence no profit for leaving the asset as collateral

[the remark this answers is not in the document]
Comment (MerlinEgalite) [reply src/moneymarkets/morpho/MorphoBlueMoneyMarketView.sol:86]: Oh ok now I understand thx

**Pull request merged**